samcli/lib/remote_invoke/sqs_invoke_executors.py

```diff
@@ -0,0 +1,146 @@
+
+    def validate_action_parameters(self, parameters: dict) -> None:
+        """
+        Validates the input boto parameters and prepares the parameters for calling the API.
+
+        Parameters
+        ----------
+        parameters: dict
+            Boto parameters provided as input
+        """
+        try:
+            for parameter_key, parameter_value in parameters.items():
+                if parameter_key == QUEUE_URL:
+                    LOG.warning("QueueUrl is defined using the value provided for resource_id argument.")
+                elif parameter_key == MESSAGE_BODY:
+                    LOG.warning(
+                        "MessageBody is defined using the value provided for either --event or --event-file options."
+                    )
+                elif parameter_key == DELAY_SECONDS:
+                    self.request_parameters[parameter_key] = int(parameter_value)
+                elif parameter_key in {MESSAGE_ATTRIBUTES, MESSAGE_SYSTEM_ATTRIBUTES}:
+                    self.request_parameters[parameter_key] = json.loads(parameter_value)
+                else:
+                    self.request_parameters[parameter_key] = parameter_value
+        except (ValueError, JSONDecodeError) as err:
+            raise InvalidResourceBotoParameterException(f"Invalid value provided for parameter {parameter_key}", err)
+
+    def _execute_action(self, payload: str) -> RemoteInvokeIterableResponseType:
+        """
+        Calls "send_message" method to send a message to the SQS queue.
+
+        Parameters
+        ----------
+        payload: str
+            The MessageBody which will be sent to the SQS
+
+        Yields
+        ------
+        RemoteInvokeIterableResponseType
+            Response that is consumed by remote invoke consumers after execution
+        """
+        if payload:
+            self.request_parameters[MESSAGE_BODY] = payload
+        else:
+            self.request_parameters[MESSAGE_BODY] = "{}"
+        self.request_parameters[QUEUE_URL] = self._queue_url
+        LOG.debug(
+            "Calling sqs_client.send_message with QueueUrl:%s, MessageBody:%s",
+            self.request_parameters[QUEUE_URL],
+            payload,
+        )
+        try:
+            send_message_response = cast(dict, self._sqs_client.send_message(**self.request_parameters))
+
+            if self._remote_output_format == RemoteInvokeOutputFormat.JSON:
+                yield RemoteInvokeResponse(send_message_response)
+            if self._remote_output_format == RemoteInvokeOutputFormat.TEXT:
+                # Create an object with MD5OfMessageBody and MessageId fields, and write to stdout
+                md5_of_message_body = send_message_response.get("MD5OfMessageBody", "")
+                message_id = send_message_response.get("MessageId", "")
+                md5_of_message_attributes = send_message_response.get("MD5OfMessageAttributes", "")
+                if md5_of_message_body and message_id:
```

Comment:
Is it OK for not yielding anything from the function that returns an iterator?

Reply:
Updated the function in the latest changes to always yield something as `MessageId` and `MD5OfMessageBody` will always be present if the send_message API call does not fail.